util/llbutil/authprovider/cloudauth/ecr_login.go

```diff
@@ -0,0 +1,70 @@
+package cloudauth
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/awslabs/amazon-ecr-credential-helper/ecr-login/api"
+	"github.com/docker/cli/cli/config/types"
+)
+
+func (ap *authProvider) getAuthConfigECR(ctx context.Context, fullPathPrefix, pathPrefix, org, project, host string) (*authConfig, error) {
+	registry, err := api.ExtractRegistry(host)
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract aws ecr registry details from %s: %w", host, err)
+	}
+
+	if registry.FIPS {
+		return nil, fmt.Errorf("FIPS registry %s not supported", host)
+	}
+
+	accessKeyIDPath := pathPrefix + "AWS_ACCESS_KEY_ID"
+	secretAccessKeyPath := pathPrefix + "AWS_SECRET_ACCESS_KEY"
```

Comment:
The term `pathPrefix` makes me think we should be using `path.Join` here, to ensure that we get a `/` separator.

Reply:
we need `/` even if running under windows; but the majority of this code has been changed in the latest commit.

Reply:
edit.... looks like path.Join does that; whereas filepath.Join might be a `\\`.

Reply:
`path` always uses `/` (even on windows). `filepath` is the one that changes the separator character depending on OS. Think of it like: `path` is intended for URL paths, `filepath` is intended for local filesystem paths. In this case, `path` should be correct in all scenarios.